frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1055,6 +1058,51 @@ pub mod pallet {
 			<QueuedSolution<T>>::put(solution);
 			Ok(())
 		}
+
+		// TODO: Add Proper Benchmarks
+		#[pallet::weight(100)]
+		pub fn challenge_solution(origin: OriginFor<T>, index: u32) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			if let Some(submission) = SignedSubmissionsMap::<T>::get(index) {
+				ensure!(
+					T::Currency::can_slash(&who, submission.deposit),
+					<Error<T>>::NotEnoughChallengerDeposit
+				);
+				match Self::feasibility_check(
+					submission.raw_solution.clone(),
+					ElectionCompute::Signed,
+				) {
+					Ok(solution) => {
+						let _ = T::Currency::slash(&who, submission.deposit);
+						<QueuedSolution<T>>::put(solution);
+						SignedSubmissionsMap::<T>::remove(index);
```

Comment:
This is likely wrong: You need to remove it from `SignedSubmissionsMap` and `SignedSubmissionsIndices`. Exactly to prevent this, we have `SignedSubmissions` wrapper.

Reply:
I would need to introduce a new method `pop` to `SignedSubmissions` wrapper that removes the values of `indices` & `insertion_overlay` fields?

Reply:
Correct.

Reply:
I was wrong here. Your approach is a bit flawed from the ground up. Please NEVER read any of these storage maps directly, and only work with the `SignedSubmissions` wrapper.